### ISSUE TITLE
ArangoDB server shutdown unexpectedly.

### ISSUE THREAD

**Issue by lkevin0** [arangodb/arangodb#3044]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [x] 3.2.1
- [ ] self-compiled devel branch

Mode:
- [ ] Cluster
- [x] Single-Server

Storage-Engine:
- [ ] mmfiles
- [x] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [x] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [ ] web interface with this browser:     running on this OS:
- [x] arangosh
- [ ] this Driver:

### These are the steps to reproduce:
1. Create collection using createCollection.js via arangosh
```
arangosh --javascript.execute createCollection.js
```
2. Import data
```
arangoimp --collection transactions --file data.json
```
3. Execute a query
```
db._query(`
FOR t IN transactions
  COLLECT accountId = t.accountId INTO g
  FILTER LENGTH(g) > 3
  LET previous = SLICE(g[*].t.status,-4, 3)
  COLLECT p = previous INTO G
RETURN DISTINCT {
  "pattern": p,
  "count": LENGTH(FOR i IN G[*].g[*].t.status FILTER LAST(i) == "E" OR LAST(i) == "F" OR LAST(i) == "G" RETURN i),
  "total": LENGTH(G),
  "accountIds": UNIQUE(G[*].g[*].t.accountId[**]),
  "subPatterns": (FOR i IN G[*].g[*].t.status RETURN DISTINCT SLICE(i,-4, 4))
}`);
```
Got ArangoError 2001: Connection closed by remote
and ArangoDB Server shut itself down immediately.

Script and data:
[arango.zip](https://github.com/arangodb/arangodb/files/1227728/arango.zip)

**Comment by jsteemann** [comment]
Fixed in 3.2.2

**Comment by dothebart** [comment]
ArangoDB 3.2.2 containing this bugfix is available for download - thanks for reporting.

Closing.
